packages/grafana-ui/src/components/PanelChrome/PanelChrome.tsx

```diff
@@ -21,7 +21,7 @@ import { TitleItem } from './TitleItem';
  */
 export interface PanelChromeProps {
```

Comment:
the issue with the changes in this file is that `height` will always get passed to the children as number (0 when the `height` props is not defined), while it may make more sense to pass `undefined` down is such cases so.

ie.

```ts
<PanelChrome >
{(width, height) => {
  // here height should be undefined because it is unknown, right no it's 0 which is wrong and may cause issues.
})
</PanelChrome>
```

and
```ts
<PanelChrome height={100} >
{(width, height) => {
  // here height is a number given the height is fixed an known
})
</PanelChrome>
```

cc. @joshhunt i tried briefly making `PanelChrome` generic but had to jump on other things so i don't have an example implementation, wdyt?

Reply:
Can we allow a React node as child in this scenario? (When size by content is enabled)

Reply:
only the height is dynamic tho, width is still fixed (and required) and should be passed in the render function. i think if we want a react node as a child it should only happen when both height and width are dynamic

Reply:
One could imagine a Table visualization with many columns and it should only take up the width needed for the columns etc, so limiting the size by content to only height seems to make it more complex.

Reply:
yeah, that makes sense 👍 

@harisrozajac lmk if you want to pair on this

Reply:
I've addressed this and now we can either pass width and height or if we don't pass them both the container will auto size with the content. This required changing the structure of PanelChrome prop type a little bit. Let me know what you guys think.